ts/src/base/Exchange.ts

```diff
@@ -874,58 +877,117 @@ export default class Exchange {
+
+    async loadProxyModules () {
+        // todo: possible sync alternatives: https://stackoverflow.com/questions/51069002/convert-import-to-synchronous
+        if (this.httpProxyAgentModule === undefined) {
+            this.httpProxyAgentModule = await import (/* webpackIgnore: true */ '../static_dependencies/proxies/http-proxy-agent/index.js');
+        }
+        if (this.httpsProxyAgentModule === undefined) {
+            this.httpsProxyAgentModule = await import (/* webpackIgnore: true */ '../static_dependencies/proxies/https-proxy-agent/index.js');
+        }
+        if (this.socksProxyAgentModule === undefined && this.socksProxyAgentModuleChecked === false) {
+            this.socksProxyAgentModuleChecked = true;
+            try {
+                // @ts-ignore
+                this.socksProxyAgentModule = await import (/* webpackIgnore: true */ 'socks-proxy-agent');
+            } catch (e) {}
+        }
+    }
+
+    setProxyAgents (httpProxy, httpsProxy, socksProxy) {
+        if (httpProxy) {
+            if (this.httpProxyAgentModule === undefined) {
+                throw new NotSupported (this.id + ' you need to initialize proxies with `.loadProxyModules()` method at first to use proxies');
+            }
+            if (!(httpProxy in this.proxyDictionaries)) {
+                this.proxyDictionaries[httpProxy] = new this.httpProxyAgentModule.HttpProxyAgent(httpProxy);
+            }
+            this.agent = this.proxyDictionaries[httpProxy];
+        } else if (httpsProxy) {
+            if (this.httpsProxyAgentModule === undefined) {
+                throw new NotSupported (this.id + ' you need to initialize proxies with `.loadProxyModules()` method at first to use proxies');
+            }
+            if (!(httpsProxy in this.proxyDictionaries)) {
+                this.proxyDictionaries[httpsProxy] = new this.httpsProxyAgentModule.HttpsProxyAgent(httpsProxy);
+            }
+            this.agent = this.proxyDictionaries[httpsProxy];
+            this.agent.keepAlive = true;
+        } else if (socksProxy) {
+            if (this.socksProxyAgentModule === undefined) {
+                throw new NotSupported (this.id + ' - to use SOCKS proxy with ccxt, at first you need install module "npm i socks-proxy-agent" and then initialize proxies with `.loadProxyModules()` method');
+            }
+            if (!(socksProxy in this.proxyDictionaries)) {
+                this.proxyDictionaries[socksProxy] = new this.socksProxyAgentModule.SocksProxyAgent(socksProxy);
+            }
+            this.agent = this.proxyDictionaries[socksProxy];
+        }
+    }
+
     async fetch (url, method = 'GET', headers: any = undefined, body: any = undefined) {
 
+        // load node-http(s) modules only on first call
+        if (isNode) {
+            if (!this.nodeHttpModuleLoaded) {
+                this.nodeHttpModuleLoaded = true;
+                const httpsModule = await import (/* webpackIgnore: true */'node:https')
```

Comment:
@ttodua didn't you say that we would do top-level imports instead or am I confusing things here?

Reply:
per your message here https://discord.com/channels/690203284119617602/1152868868293926932/1164678932264648776
> can't we await import the minified version?
so, I've done it with `await` inside clause.
the second thing is that, as it is specifically for node-js, and not for js version as a whole (because it would not work on browser) I thought we could leave it there. wdyt?